src/Compilation.zig

```diff
@@ -1446,6 +1447,13 @@ pub fn create(gpa: Allocator, options: InitOptions) !*Compilation {
             system_libs.putAssumeCapacity(lib_name, options.system_lib_infos[i]);
         }
 
+        var frameworks: std.StringArrayHashMapUnmanaged(Framework) = .{};
+        errdefer frameworks.deinit(gpa);
+        try frameworks.ensureTotalCapacity(gpa, options.framework_names.len);
+        for (options.framework_names, options.framework_infos) |framework_name, info| {
+            frameworks.putAssumeCapacity(framework_name, info);
+        }
```

Comment:
This no longer needs to be a hash map, right?

```suggestion
        const frameworks = try arena.alloc(Framework, options.framework_names.len);
        for (frameworks, options.framework_names, options.framework_infos) |*d, framework_name, info| {
            d.* = frameworks.putAssumeCapacity(framework_name, info);
        }
```

`system_libs` is a mutable hash map because `Compilation.addLinkLib` must add libraries found during the compilation process. If I understand correctly, that is not something that happens with frameworks, so frameworks can be less complicated.

If it's fine with the linker logic, `framework_names` and `framework_infos` could just be stored directly into the `link.File.Options`; no need to copy the memory another data structure.

Alternately, if a hash map is indeed desired, perhaps the CLI should actually pass a hash map rather than two arrays and then that can be used directly.

Reply:
Ahhh, makes sense, thanks for explaining and lemme make this change as it's definitely desirable!

Reply:
Done in b73ef34.